src/patterns/ProgressWizard/hooks/useHistoryManagement.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useRef } from 'react';
+
+export const useHistoryManagement = ({
```

Comment:
would be good to add jsdoc comments here for what each argument means  (i.e. `manageHistory` does) and what the overall purpose of the hook is above line 3.  Debatably `manageHistory` could be `enabled`?